File: Week 01/id_503/leetcode_283_503.go
package week01

// 双指针法
func moveZeroes(nums []int) {

	nonZeroIndex := 0
	for i := range nums {
		if nums[i] != 0 {
			nums[nonZeroIndex], nums[i] = nums[i], nums[nonZeroIndex]
			nonZeroIndex++
		}
	}
}

File: Week 01/id_503/leetcode_283_503.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
	let nonZeroIndex = 0;
	for (let i = 0; i < nums.length; i++) {
		if (nums[i] != 0) {
			const temp = nums[nonZeroIndex];
			nums[nonZeroIndex] = nums[i];
			nums[i] = temp;

			nonZeroIndex++
		}
	}
};